homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -1370,6 +1438,31 @@ async def text_to_speech(self, tts_input: str) -> None:
             PipelineEvent(PipelineEventType.TTS_END, {"tts_output": tts_output})
         )
 
+    async def acknowledge(self, media_id: str, tts_input: str | None) -> None:
+        """Respond with acknowledge media instead of text-to-speech."""
+        self.process_event(
+            PipelineEvent(
+                PipelineEventType.TTS_START,
+                {
+                    "language": self.pipeline.tts_language,
+                    "voice": self.pipeline.tts_voice,
+                    "tts_input": tts_input or "",
+                },
+            )
+        )
+
+        if media_source.is_media_source_id(media_id):
+            media = await media_source.async_resolve_media(self.hass, media_id, None)
+            media_id = media.url
+        else:
+            media_id = network.get_url(self.hass) + media_id
+
+        tts_output = {"url": media_id}
+
+        self.process_event(
+            PipelineEvent(PipelineEventType.TTS_END, {"tts_output": tts_output})
```

Comment:
hmm with streaming TTS, we're moving to getting the TTS URL from the RUN-START event. We don't use this data anymore. 

So instead, we should upgrade the [TTS ResultStream](https://github.com/home-assistant/core/blob/dev/homeassistant/components/tts/__init__.py#L469) to override the generated audio